core/src/main/clojure/xtdb/pgwire.clj

```diff
@@ -972,7 +1028,7 @@
 
     (letfn [(->cursor ^xtdb.IResultCursor [xt-args]
               (util/with-close-on-catch [args-rel (vw/open-args allocator xt-args)]
-                (.openQuery prepared-query (assoc query-opts :args args-rel))))
+                (with-auth-check conn (.openQuery prepared-query (assoc query-opts :args args-rel)))))
```

Comment:
I'd probably stick `with-auth-check` around the with-close-on-catch here, feels like you're doing the auth check halfway through 🙂